doc/languages-frameworks/factor.section.md

```diff
@@ -0,0 +1,191 @@
+```nix
+{ lib, fetchFromGitHub, factorPackages, curl }:
+
+factorPackages.buildFactorApplication (finalAttrs: {
+  pname = "painter";
+  version = "1";
+
+  factor-lang = factorPackages.factor-minimal-gui;
+
+  src = fetchFromGitHub {
+    name = finalAttrs.pname;
+    owner = "Capital-EX";
+    repo = finalAttrs.pname;
+    rev = "365797be8c4f82440bec0ad0a50f5a858a06c1b6";
+    hash = "sha256-VdvnvKNGcFAtjWVDoxyYgRSyyyy0BEZ2MZGQ71O8nUI=";
+  };
+
+  sourceRoot = ".";
+
+  enableUI = true;
+  extraVocabs = [ factorPackages.bresenham ];
+
+  extraPaths = with finalAttrs.factor-lang; binPackages ++ defaultBins ++ [ curl ];
+
+})
+```
+
+The use of the `src.name` and`sourceRoot` attributes conveniently establish the necessary `painter` vocabulary directory that is needed for the deployment to work.
+
+It requires the packager to specify the full set of binaries to be made available at runtime.
+This enables the standard pattern for application packages to specify all runtime dependencies explicitly without the Factor runtime interfering.
+
+Additional attribute that are understood by `buildFactorApplication`:
+- `vocabName` is the path to the vocabulary to be deployed relative to the source root.
+  So, directory `foo/` from the example above could be `extra/deep/down/foo`.
+  This allows you to maintain Factor's vocabulary hierarchy and distribute the same source tree as a stand-alone application and as a library in the Factor development environment via the `extraVocabs` attribute.
+- `binName` is the name of the resulting binary in `/bin/`.
+  It defaults to the last directory component in `vocabName`.
+  It is also added as the `meta.mainProgram` attribute to facilitate `nix run`.
+- `enableUI` is `false` by default.
+  Set this to `true` when you ship a graphical application.
+- `extraLibs` adds additional libraries as runtime dependencies.
+  Defaults to `[]` and is concatenated with `runtimeLibs` from the used factor-lang package.
+  Use `factor-minimal` to minimize the closure of runtime libraries.
+- `extraPaths` adds additional binaries to the runtime PATH environment variable (without adding their libraries, as well).
+  Defaults to `[]` and is concatenated with `defaultBins` and `binPackages` from the used factor-lang package.
+  Use `factor-minimal` to minimize the closure of runtime libraries.
+- `deployScriptText` is the actual deploy Factor file that is executed to deploy the application.
+  You can change it if you need to perform additional computation during deployment.
+- `factor-lang` overrides the Factor package to use to deploy this application, which also affects the default library bindings and programs in the runtime PATH.
+  It defaults to `factor-lang` when `enableUI` is turned on and `factor-no-gui` when it is turned off.
+  Applications that use only Factor libraries without external bindings or programs may set this to `factor-minimal` or `factor-minimal-gui`.
```

Comment:
It's not a hard rule for me personally, because documenting all that is already great in itself, but we *do* have a rule for using [definition lists for function arguments](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#function-arguments) -- that will help migrating whatever is done to different formats programmatically.

Sorry for picking things piecemeal. I'm out again, I primarily just monitor how the redirects system is being picked up and whether it produces excess friction.

Reply:
Thanks, I addressed your comments.